autoscale/src/test/java/com/sequenceiq/periscope/endpointtests/DistroXAutoScaleScalingActivityV1EndpointTest.java

```diff
@@ -0,0 +1,484 @@
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto;
+import com.sequenceiq.authorization.service.ResourceAuthorizationService;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.StackType;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.cloudbreak.auth.altus.UmsClient;
+import com.sequenceiq.cloudbreak.quartz.configuration.QuartzJobInitializer;
+import com.sequenceiq.periscope.api.endpoint.v1.DistroXAutoScaleScalingActivityV1Endpoint;
+import com.sequenceiq.periscope.api.model.ActivityStatus;
+import com.sequenceiq.periscope.api.model.ClusterState;
+import com.sequenceiq.periscope.api.model.DistroXAutoscaleScalingActivityResponse;
+import com.sequenceiq.periscope.client.AutoscaleUserCrnClientBuilder;
+import com.sequenceiq.periscope.domain.Cluster;
+import com.sequenceiq.periscope.domain.ClusterPertain;
+import com.sequenceiq.periscope.domain.ScalingActivity;
+import com.sequenceiq.periscope.repository.ClusterPertainRepository;
+import com.sequenceiq.periscope.repository.ClusterRepository;
+import com.sequenceiq.periscope.repository.ScalingActivityRepository;
+import com.sequenceiq.periscope.service.AutoscaleRestRequestThreadLocalService;
+import com.sequenceiq.periscope.service.configuration.ClusterProxyConfigurationService;
+import com.sequenceiq.periscope.testcontext.EndpointTestContext;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = EndpointTestContext.class)
+@ActiveProfiles("test")
+public class DistroXAutoScaleScalingActivityV1EndpointTest {
+
+    private static final String SERVICE_ADDRESS = "http://localhost:%d/as";
+
+    private static final Long TEST_USER_ID = 100L;
+
+    private static final Long TEST_WORKSPACE_ID = 100L;
+
+    private static final String TEST_ACCOUNT_ID = "accid";
+
+    private static final String TEST_CLUSTER_NAME = "testCluster";
+
+    private static final String TEST_CLUSTER_NAME_1 = "testCluster1";
+
+    private static final String TEST_CLUSTER_NAME_2 = "testCluster2";
+
+    private static final String TEST_CLUSTER_NAME_NULL = null;
+
+    private static final String TEST_TENANT = "testTenant";
+
+    private static final String TEST_USER_CRN = String.format("crn:cdp:iam:us-west-1:%s:user:mockuser@cloudera.com", TEST_ACCOUNT_ID);
+
+    private static final String TEST_CLUSTER_CRN = String.format("crn:cdp:iam:us-west-1:%s:cluster:mockuser@cloudera.com", TEST_ACCOUNT_ID);
+
+    private static final String TEST_CLUSTER_CRN_1 = String.format("crn:cdp:iam:us-west-1:%s:cluster:mockuser@cloudera.com", "aacid1");
+
+    private static final String TEST_CLUSTER_CRN_2 = String.format("crn:cdp:iam:us-west-1:%s:cluster:mockuser@cloudera.com", "aacid2");
+
+    private static final String TEST_CLUSTER_CRN_NULL = null;
+
+    private static final String TEST_ACTIVITY_CRN = "crn:cdp:autoscale:us-west-1:9d74eee4-1cad-45d7-b645-7ccf9edbb73d:datahubAutoscaleActivity:6d" +
```

Comment:
This could have been an array and a loop in a static block to populate it. That typically makes writing tests easier as well - to be able to refer to entries programatically.
Is ok for this patch.